Packs/Rundeck/Integrations/Rundeck/Rundeck.py

```diff
@@ -0,0 +1,1113 @@
+        )
+
+    def adhoc_script_run(self, project_name: str, arg_string: str, node_thread_count: str, node_keepgoing: str, as_user: str, node_filter: str
+                                     ,script_interpreter: str, interpreter_args_quoted: str, file_extension: str, entry_id: str):
+        """
+        This function runs a script from file
+        :param project_name: project to run the script file
+        :param arg_string: arguments for the script when executed
+        :param node_thread_count: threadcount to use
+        :param node_keepgoing: 'true' for continue executing on other nodes after a failure. false otherwise
+        :param as_user: identifying the user who ran the job
+        :param node_filter:
+        :param script_interpreter: a command to use to run the script
+        :param interpreter_args_quoted: if true, the script file and arguments will be quoted as the last argument to
+        :param file_extension: extension of of the script file
+        :param entry_id: Demisto id for the uploaded script file you want to run
+        :return: api response
+        """
+
+        request_params: Dict[str, any] = {}
+        if arg_string:
+            request_params['argString'] = arg_string
+        if node_thread_count:
+            request_params['nodeThreadcount'] = node_thread_count
+        if node_keepgoing:
+            request_params['nodeKeepgoing'] = node_keepgoing
+        if as_user:
+            request_params['asUser'] = as_user
+        if script_interpreter:
+            request_params['scriptInterpreter'] = script_interpreter
+        if interpreter_args_quoted:
+            request_params['interpreterArgsQuoted'] = interpreter_args_quoted
+        if file_extension:
+            request_params['fileExtension'] = file_extension
+        if node_filter:
+            request_params['filter'] = node_filter
+        if project_name:
+            project_name_to_pass = project_name
+        else:
+            project_name_to_pass = self.project_name
+
+        file_path = demisto.getFilePath(entry_id).get("path", None)
+        if not file_path:
+            raise DemistoException(
+                f"Could not find file path to the next entry id: {entry_id}. \n"
+                f"Please provide another one."
+            )
+        else:
+            file_name = ntpath.basename(file_path)
+
+        request_params.update(self.params)
+        del self._headers['Content-Type']
+        with open(file_path, "rb") as file:
+            self._headers.update({'Content-Disposition': f'form-data; name="file"; filename="{file_name}"'})
+            return self._http_request(
+                method="POST",
+                files={'scriptFile': file},
+                url_suffix=f'/project/{project_name_to_pass}/run/script',
+                params=request_params
+            )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def filter_results(results: Union[list, dict], fields_to_remove: list, remove_signs) -> Union[list, dict]:
```

Comment:
please remove logs from this function. this is an internal object manipulation function, no need for logging.
as long as its not errors. then we will want to add the input arguments in the error.